typing/typecore.ml

```diff
@@ -2557,22 +2549,22 @@ let contains_polymorphic_variant p =
   in
   try loop p; false with Exit -> true
 
-let contains_gadt env p =
-  let rec loop env p =
+let contains_gadt p =
```

Comment:
> Before it would check if some constructors in the pattern where part of a "GADT type"

That's not what `lookup_all_constructors` does. It looks up all the constructors in the environment with the same name. So the check used to return `true` if any of the constructors with that name in scope was a GADT. The function now operates on the typedtree, after disambiguation, and so already knows which of the constructors was selected.